providers/implementations/keymgmt/ml_kem_kmgmt.c.in

```diff
@@ -482,7 +482,12 @@ static int ml_kem_import(void *vkey, int selection, const OSSL_PARAM params[])
     if (res > 0 && include_private
         && !ml_kem_pairwise_test(key, key->prov_flags)) {
 #ifdef FIPS_MODULE
-        ossl_set_error_state(OSSL_SELF_TEST_TYPE_PCT);
+        if (ossl_fips_self_testing())
+            ossl_set_error_state(OSSL_SELF_TEST_TYPE_PCT);
+        else
+            ERR_raise_data(ERR_LIB_PROV, PROV_R_INVALID_KEY,
+                           "explicit %s public key does not match private",
+                           ossl_ml_kem_key_vinfo(key)->algorithm_name);
```

Comment:
There is no need for any checking here.  During power up testing, it cannot fail.
```suggestion
        ERR_raise_data(ERR_LIB_PROV, PROV_R_INVALID_KEY,
                       "explicit %s public key does not match private",
                       ossl_ml_kem_key_vinfo(key)->algorithm_name);
```

I'd even go as far as removing the `#ifdef FIPS_MODULE` wrapping since this error is useful regardless.